### PR TITLE
Add parameter_filter to arguments in backtrace

### PR DESCRIPTION
**Pull request by hiphapis**
Currently, parameter is print without `Rails.application.config.filter_parameters`.
So i did  modify this part

AS IS
```
Loc  | Field                | Object | Arguments                                    | Result
21:9 | Mutations.signInUser | nil    | {"email"="test@test.com", "password"="1234"} | ~~~
```
TO BE
```
Loc  | Field                | Object | Arguments                                          | Result
21:9 | Mutations.signInUser | nil    | {"email"="test@test.com", "password"="[FILTERED]"} | ~~~
```

### CODE REVIEW

File: lib/graphql/backtrace/table.rb [hunk 1]
```diff
@@ -36,6 +36,13 @@ def to_backtrace
 
       private
 
+      def parameter_filter(params)
+        return params if Rails.application.config.filter_parameters.blank?
```
Comment (swalkinshaw) [inline lib/graphql/backtrace/table.rb]: graphql-ruby doesn't have a dependence on Rails so `Rails` and `ActionDispatch` can't be referenced like this

Comment (hiphapis) [reply lib/graphql/backtrace/table.rb]: Ah. I see
So you mean, need to check exists Rails or not?
Hm... I guess add some configuration is better. but `graphql-ruby` doesn't have a configuration. isn't it?

Which way is better?

Comment (swalkinshaw) [reply lib/graphql/backtrace/table.rb]: It's not going to be easy for the reason you mentioned... You'd also want this behaviour to be optional as well since not everyone would want it filtered by default.

**Comment by swalkinshaw** [comment]
I'm guessing this is because you use the backtrace feature in production? I'm assuming a lot of people only have it enabled in development (Shopify included).

So before you put in more work on this feature maybe @rmosolgo can chime in if its a feature he's interested in.

**Comment by hiphapis** [comment]
@swalkinshaw Production mode also still using this backtrace
Production used when meet Fatal Error log, like this image
image

File: lib/graphql/backtrace/table.rb (hunk 1 above)
Comment (hiphapis) [reply lib/graphql/backtrace/table.rb]: Yes. right. so i suggest "put to configuration". then user can set filters with env.  but unfortunately, don't have a configuration

**Comment by hiphapis** [comment]
I concerned about this. and I guess have 2 ways
First thing is using parameter of `GraphQL::Backtrace`
For example,
```ruby
filter_parameters = [:password]
GraphQL::Backtrace(filter_parameters: filter_parameters)
```
like this

Second thing is make and using `filter_parameters`
For example,
```ruby
TestSchema = GraphQL::Schema.define do
filter_parameters([:password])
```
like this.

How about this?

**Comment by swalkinshaw** [comment]
The 2nd option isn't really specific to the backtrace feature. It could be renamed to be more specific at least (`backtrace_filter_parameters`) but it's still a global schema option for something that's designed as a plugin.

To me the first option makes more sense since it's a backtrace-only setting. However, wouldn't you still need to specify *what* is going to the filter the parameters? `GraphQL::Backtrace(filter_parameters: [:password])` still assumes it's using `ActionDispatch::Http::ParameterFilter` under the hood.

You could do:

```ruby
GraphQL::Backtrace(filter_parameters: [:password], filter_class: ActionDispatch::Http::ParameterFilter)
```

Basically anything that responds to `filter` could be used.

I should note the backtrace feature is still marked as "Experimental" as well so I'm not sure how @rmosolgo feels about adding features to it.

**Comment by hiphapis** [comment]
@swalkinshaw I see. and I thought, this gem does not depend on `Rails` so need to make filter function manually as like' ActionDispatch::Http::ParameterFilter'.
If concept is decide then i will try to like that

**Comment by rmosolgo** [comment]
It looks like this has stalled, so I'll close it. 

As @swalkinshaw mentioned, the backtrace helper is really only meant for development. IIRC it adds some nontrivial overhead to execution because of its bookkeeping.
